artemis-server/src/main/java/org/apache/activemq/artemis/core/server/Consumer.java

```diff
@@ -21,7 +21,7 @@
 import org.apache.activemq.artemis.core.filter.Filter;
 import org.apache.activemq.artemis.spi.core.protocol.SessionCallback;
 
-public interface Consumer {
+public interface Consumer extends PriorityAware {
```

Comment:
Just a design q: why using a specific `PriorityAware` interface?
I'm assuming that we can't have anymore a `Consumer` that doesn't provide a `default int getPriority()`

Reply:
Correct, but to make testing of the QueueConsumersImpl logic easier and quite self contained this, its kept seperate, also it means if we wanted to reuse the new QueueConsumerImpl for other PriorityAware needs it makes it easier.